python/ray/air/_internal/torch_utils.py

```diff
@@ -190,3 +190,19 @@ def load_torch_model(
             f"to be of type `torch.nn.Module`, or a model "
             f"state dict of type dict."
         )
+
+
+def contains_tensor(obj):
```

Comment:
Do we need this? I think if torch is installed, it should be safe to always use the TorchBackend for encoding/decoding (even if the data dict does not contain a tensor). I'm worried in the worst case `contains_tensor` can lead to a lot of recursion.

Reply:
Turns out we do need it as `torch.save` seems to silently fail if a full model is passed (and not a state dict).

I think it should be fine - a similar lookup has to be in pickling after all, and in most cases it should finish early.